tru_tags.php

```diff
@@ -333,18 +333,27 @@ function tru_tags_cloud_query($atts) {
 	$filter = tru_tags_filter_sections($excludesection);
 	$filter .= ($filtersearch ? filterSearch() : '');
 
+	// Using SQL caching capabilities. Txp 4.6+
+	if (function_exists('now')) {
+		$where_posted = now('posted');
+		$where_expired = " AND (".now('expires')." <= Expires OR Expires IS NULL)";
+	} else {
+		$where_posted = "now()";
+		$where_expired = " AND (now() <= Expires OR Expires = ".NULLDATETIME.")";
```

Comment:
In the version of Textpattern 4.6, there were changes in `textpattern` table in the `expires` field.

* Txp 4.6: [Field `expires`  change](https://github.com/textpattern/textpattern/blob/dev/textpattern/update/_to_4.6.0.php#L215)
* Txp 4.6: [Updating existing data](https://github.com/textpattern/textpattern/blob/dev/textpattern/update/_to_4.6.0.php#L231)  from `0000-00-00` (aka NULLDATETIME) to `NULL`

So for 4.6+, use `IS NULL`, and for earlier versions, use `NULLDATETIME`.